s3/s3.go

```diff
@@ -201,14 +206,14 @@ func (b *Bucket) GetReader(path string) (rc io.ReadCloser, err error) {
 
 // PutReader inserts an object into the S3 bucket by consuming data
 // from r until EOF.
-func (b *Bucket) PutReader(path string, r io.Reader, length int64, contType string, perm ACL) error {
+func (b *Bucket) PutReader(path string, r io.ReadSeeker, length int64, contType string, perm ACL) error {
```

Comment:
Hrm. This is an API break, because someone could be relying on the function signature (e.g. taking a method value).